packages/wallet-manager/src/WalletManagerController.ts

```diff
@@ -0,0 +1,155 @@
+import { ReactiveControllerHost } from 'lit';
+import { EvmWallet, SubstrateWallet } from '.';
+import { Web3Provider } from '@ethersproject/providers';
+import { ApiPromise } from '@polkadot/api';
+import { AddChain } from './types';
+import { customEVMEvents, IWalletManagerController } from './interfaces';
+import { Signer } from '@ethersproject/abstract-signer';
+
+export class WalletManagerController implements IWalletManagerController {
+  private host: ReactiveControllerHost;
+  evmWallet?: EvmWallet;
+  substrateWallet?: SubstrateWallet;
+  account?: string;
+  substrateAccount?: string;
+
+  constructor(
+    host: ReactiveControllerHost,
+    initArguments?: {
+      web3Provider?: Web3Provider;
+      apiPromise?: ApiPromise;
+      wssConnectionUrl?: string;
+    }
+  ) {
+    (this.host = host).addController(this);
+
+    if (initArguments?.web3Provider) {
+      this.initWeb3Provider(initArguments.web3Provider);
+    } else if (initArguments?.apiPromise) {
+      this.connectFromApiPromise(initArguments.apiPromise);
+    } else if (initArguments?.wssConnectionUrl) {
+      this.connectFromWssProvider(initArguments.wssConnectionUrl);
+    } else {
+      this.initWeb3Provider();
+    }
+  }
```

Comment:
IMO it'd more readable and semantic if constructor got arguments `type` (EVM or Substrate, these can be in enum) and optional `initParam` of type `Web3Provider | ApiPromise | string`, and then depending on the `type`, we have `initEvmWallet` which inits from provider or window and `initSubstrateWallet` which, if initParam is string, connects from wss provider or, if initParam isn't string,  connects from ApiPromise.